src/scores/continuous/interval_impl.py

```diff
@@ -0,0 +1,100 @@
+            if neither are supplied is to reduce all dims.
+        preserve_dims: Optionally specify which dimensions to preserve when calculating
+            quantile interval score. All other dimensions will be reduced. As a special case,
+            'all' will allow all dimensions to be preserved. In this case, the result will be in
+            the same shape/dimensionality as the forecast, and the errors will be the quantile
+            score at each point (i.e. single-value comparison against observed), and the
+            forecast and observed dimensions must match precisely. Only one of `reduce_dims`
+            and `preserve_dims` can be supplied. The default behaviour if neither are supplied
+            is to reduce all dims.
+        weights: Optionally provide an array for weighted averaging (e.g. by area, by latitude,
+            by population, custom)
+
+    Returns:
+        A Dataset with the dimensions specified in `dims`.
+        The dataset will have the following data variables:
+
+        - interval_width_penalty: fcst_upper_qtile - fcst_lower_qtile
+        - overprediction_penalty: ((1/lower_qtile_level)*(fcst_lower_qtile-obs)).clip(min=0)
+        - underprediction_penalty: ((1/upper_qtile_level)*(fcst_upper_qtile-obs)).clip(min=0)
+        - total: sum of all penalties
+
+    Raises:
+        ValueError: If not (0 < lower_qtile_level < upper_qtile_level < 1).
+        ValueError: If (fcst_lower_qtile > fcst_upper_qtile).any().
+
+    References:
+        T. Gneiting, and Raftery A.E., "Strictly proper scoring rules, prediction and
+        estimation", J. Amer. Stat. Assoc., Vol. 102 No. 477 (2007), pp. 359--378,
+        Corollary 5.2.
+    """
+
+    if not 0 < lower_qtile_level < upper_qtile_level < 1:
+        raise ValueError(
+            "Expected 0 < lower_qtile_level < upper_qtile_level < 1. But got "
+            f"lower_qtile_level = {lower_qtile_level} and upper_qtile_level = {upper_qtile_level}"
+        )
+    # Check fcst_lower_qtile and fcst_upper_qtile have the same dimension
+    check_dims(fcst_upper_qtile, fcst_lower_qtile.dims, mode="equal")
+    specified_dims = reduce_dims or preserve_dims
+    # check requested dims are a subset of fcst_lower_qtile (or fcst_upper_qtile) dimensions
+    if specified_dims is not None:
+        check_dims(xr_data=fcst_lower_qtile, expected_dims=specified_dims, mode="superset")
+    # check obs dimensions are a subset of fcst_lower_qtile (or fcst_upper_qtile) dimensions
+    check_dims(xr_data=obs, expected_dims=fcst_lower_qtile.dims, mode="subset")  # type: ignore
+    if (fcst_lower_qtile > fcst_upper_qtile).any():
+        raise ValueError("Input does not satisfy fcst_lower_qtile < fcst_upper_qtile condition.")
+    interval_width = fcst_upper_qtile - fcst_lower_qtile
+    lower_diff = (1 / lower_qtile_level) * (fcst_lower_qtile - obs)
+    lower_diff = lower_diff.clip(min=0.0)
```

Comment:
I'm not entirely sure why this clipping is occurring, a comment could help

Reply:
It will be obvious from the definition of the function once you can read it easily.

Reply:
I hope the revised function can provide more clarification, but this clipping is because of the conditions defined for over- and under-prediction penalties (fulfilling the indicator function).